### ISSUE TITLE
Is there a way for shellcheck to understand custom aliases?

### ISSUE THREAD

**Issue by tterranigma** [koalaman/shellcheck#1132]
#### For new checks and feature suggestions
- [X] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [X] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

I am using the [Bash Infinity](https://github.com/niieani/bash-oo-framework) framework which provides some aliases, like `[array]`. I use them in the screenshot and code below. In the screenshot I am using vim with Syntastic that reads the output of shellcheck and annotates the problematic lines. While the code works fine, shellcheck complains as it cannot understand the custom aliases. Is there a way to load the aliases in shellcheck or sth to avoid what happens?

#### Here's a snippet or screenshot that shows the problem:

![screenshot_20180226_190945](https://user-images.githubusercontent.com/11805218/36859174-d4753f2e-1d7d-11e8-8371-8563fce3ebb4.png)

```sh
#!/usr/bin/env bash

function ansible_provision() {
  [array] tags=(all)
  [array] skip_tags
  [array] limit=()

  . "$_GO_USE_MODULES" 'instance'

  ansible-playbook \
    --tags="${tags[@]}" \
    --skip-tags="${skip_tags[@]}" \
    --limit="${limit[@]}" \
    -- "$PROJECT_DIR"/playbook.yml >&${STDOUT}
}
```

#### Here's what shellcheck currently says:
```
syntax error near unexpected token `('
```
If I remove the lines with the `()` in the end, I get (for the remaining `[array] skip_tags` line):

```
You need a space after the [ and before the ]. [SC1035]
```

#### Here's what I wanted or expected to see:
No errors, perhaps some warnings about other issues with the code.

**Comment by Dmole** [comment]
Would

    # shellcheck source=/opt/Infinity.sh

solve this?

**Comment by koalaman** [comment]
There's currently no way to do so, no :/

Aliases are a particular annoyance for ShellCheck because they modify subsequent parsing. For example, `((RANDOM%2)) && alias forever='while true'` followed by `forever; do sleep 1; done` will cause a bash syntax error 50% of the times you run the script. They're also rarely used in scripts at all, being disabled by default and all, so there's not much in the way of support at the moment.

**Comment by nkakouros** [comment]
I was thinking, could there be an option to define mappings to "resolve" aliases? For instance, if I have:

```
alias import='source ...more_stuff_here...'
```

a mapping could look like:

```
import = source # or whatever
```
.
